core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -140,6 +139,7 @@ public void initialize(String name, Map<String, String> properties) {
       if (initializeCatalogTables) {
         initializeCatalogTables();
       }
+      checkAndMaybeUpdateSchema();
```

Comment:
I think it is odd that this always initializes the catalog with view support when not all clients support views. Instead of an upgrade flag, I would expect this to be configured by a schema version using something like `schema.version={0,1}` that defaults to 0. Then initialization would always know which table to create and upgrading would be done with `schema.version` changes.

Reply:
Is it not problematic in terms of backward compatibility and impact of our users ? That's the reason why I tried to limit the change.

Reply:
I think it would just be better if we were explicit about the version. I think this creates a loophole where people can hit the errors that we are trying to avoid (older clients failing on views) when they initialize without making a conscious choice.

Reply:
OK, I will then replace `add-view-support` by `schema.version`. Thanks.